Sources/HTTP/Messages/PeerAddress.swift

```diff
@@ -0,0 +1,40 @@
+/// Represents the information we have about the remote
+/// peer of this message.
+/// The peer (remote/client) address is important
+/// for availability (block bad clients by their IP)
+/// or even security.
+/// We can always get the remote IP of the connection
+/// from the Stream. However, when clients go through
+/// a proxy or a load balancer, we'd like to get the
+/// original client's IP. Most proxy servers and load
+/// balancers communicate the information about the
+/// original client in certain headers.
+/// See https://en.wikipedia.org/wiki/X-Forwarded-For
+public struct PeerAddress {
+    
+    /// A newer version, sent by other proxies, only use
+    /// if you can trust the proxy (easily spoofed).
+    public let forwarded: String?
+    
+    /// Sent by certain proxies, only use if you can
+    /// trust the proxy (easily spoofed).
+    public let xForwardedFor: String?
+    
+    /// The stream (socket) remote address, pulled out
+    /// of the system. More difficult (but possible) to
+    /// spoof.
+    public let stream: String
+    
+    public init(forwarded: String? = nil, xForwardedFor: String? = nil, stream: String) {
+        self.forwarded = forwarded
+        self.xForwardedFor = xForwardedFor
```

Comment:
We should add `X-Forwarded-Proto` as well, to be able to get the scheme, when served through a load balancer with TLS Termination